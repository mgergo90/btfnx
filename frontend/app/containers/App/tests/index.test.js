import React from 'react';
import { shallow } from 'enzyme';
import { Route } from 'react-router-dom';

import { App } from '../index';

jest.mock('react', () => {
  // eslint-disable-next-line no-shadow
  const React = jest.requireActual('react');
  React.Suspense = ({ children }) => children;
  return React;
});

describe('<App />', () => {
  it('should not render routes while initialize runs', () => {
    const renderedComponent = shallow(<App classes={{}} initialize />);
    expect(renderedComponent.find(Route)).toHaveLength(0);
  });

  it('should render routes after initialize finished', () => {
    const renderedComponent = shallow(<App classes={{}} initialize={false} />);
    expect(renderedComponent.find(Route)).not.toHaveLength(0);
    expect(renderedComponent).toMatchSnapshot();
  });
});
