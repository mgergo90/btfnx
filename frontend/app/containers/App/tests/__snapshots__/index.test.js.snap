// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render routes after initialize finished 1`] = `
<Fragment>
  <HelmetWrapper
    defaultTitle="Demo"
    defer={true}
    encodeSpecialCharacters={true}
    titleTemplate="%s - Demo"
  />
  <Component
    fallback={<WithStyles(Loader) />}
  >
    <div>
      <Connect(WithStyles(Header))
        initialize={false}
      />
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/login"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/registration"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path=""
        />
      </Switch>
    </div>
  </Component>
</Fragment>
`;
